spec/views/post_index_spec.rb

```diff
@@ -0,0 +1,44 @@
+require 'rails_helper'
+
+RSpec.feature 'User posts index page', type: :feature do
+  include ActionView::Helpers::TextHelper
+
+  let(:user) { User.create(name: 'John Doe', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Lorem ipsum', posts_counter: 5) }
+
+  scenario 'User sees user details on index page' do
+    visit user_posts_path(user)
+
+    expect(page).to have_css("img[src*='#{user.photo}']")
+    expect(page).to have_content(user.name)
+    expect(page).to have_content("Number of posts: #{user.posts_counter}")
+  end
+
+  scenario 'User sees post details on index page' do
+    post = Post.create(author: user, title: 'Test Post', text: 'This is a test post.')
+
+    visit user_posts_path(user)
+
+    expect(page).to have_content(post.title)
+    expect(page).to have_content(truncate(post.text, length: 100))
+
+    expect(page).to have_content("Comments: #{post.comments_counter}")
+    expect(page).to have_content("Likes: #{post.likes_counter}")
+  end
+
+  scenario 'User sees pagination section on index page' do
+    3.times { Post.create(author: user, title: 'Test Post', text: 'This is a test post.') }
+
+    visit user_posts_path(user)
+
+    expect(page).to have_css('.post-container', count: 2)
+  end
+
+  scenario 'User redirects to post show page when clicking on a post' do
```

Comment:
- [ ] Same for this file, the test is yet to be complete as pointed in the image 
![image](https://github.com/lily4178993/nell-blog/assets/86200356/06ced311-ed0b-498b-8dce-e22270e04a20)
![image](https://github.com/lily4178993/nell-blog/assets/86200356/bbc125ba-3313-4983-bceb-9bf158437b2b)